mono_repo/lib/src/root_config.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+import 'dart:io';
+
+import 'package:json_annotation/json_annotation.dart';
+import 'package:path/path.dart' as p;
+import 'package:yaml/yaml.dart' as y;
+
+import 'package_config.dart';
+import 'user_exception.dart';
+import 'utils.dart';
+
+class RootConfig extends MapBase<String, PackageConfig> {
```

Comment:
Does this not have a consistent structure? (why not json_serializable?)

Reply:
This is a synthetic data structure composing `mono_config.yaml` + all of the `pkg_config.yaml` files. So it's not like we're parsing a single input file..

Reply:
hmm, name is super misleading then imo, PackageConfigs or something? It looks like its literally just the package configs right now right not the mono_config as well?